### DISCUSSION TITLE
Add ability to set the default orderBy in nested element indexes

### DISCUSSION THREAD

**Discussion by ryansupercool** [craftcms/cms#16236] [Ideas] [answered]
We may be missing something but there doesn't appear to be a way to set the default orderBy within nested element? (We're using the `ElementIndexViewMode::Table` view.

We've defined `defineSortOptions` which has one attribute but Craft looks to be using the ID instead. Defining `defineSources` with a "*" source and `defaultSort` also does not appear to work.

The only way that I can see of setting a default sort order is via the `sortOrder` column in `elements_owners`. For nested elements that aren't manually "sortable" is would be really helpful to be able to define the default sort via other means than having to maintain `sortOrder` for date attributes - this itself may not be the most efficient as from what I can tell you'd have to do this via `NestedElementManager::EVENT_AFTER_SAVE_ELEMENTS` which will have happened after Craft has already resorted them.

**Comment by brandonkelly** [comment, marked as answer]
Thanks for pointing this out. Craft doesn’t have any built-in embedded element indexes that aren’t user-sortable, so this was overlooked.

I just added support for setting a `defaultSort` key on the `$config` array when calling `craft\elements\NestedElementManager::getIndexHtml()` for Craft 5.6 (f613051b0af4df20d02cbd29e3f09579d9575b54), which you can set to any sortable attribute name supported by your element type’s `defineSortOptions()` method.

```php
return $nestedElementManager->getIndexHtml($owner, [
    'defaultSort' => 'attributeName',
    // ...
]);
```

It will use the sort direction specified by the sort option’s `defaultDir` value (`asc` by default). Or you can pass an array which also includes the sort direction.
```php
return $nestedElementManager->getIndexHtml($owner, [
    'defaultSort' => ['attributeName', 'desc'],
    // ...
]);
```

**Reply by ryansupercool** [reply]
I've just given this a go and it works perfectly. Thanks!

**Reply by brandonkelly** [reply]
Craft 5.6.0 is out now 🚀
